routers/api/v1/repo/languages.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package repo
+
+import (
+	"net/http"
+
+	repo_model "code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/modules/context"
+	api "code.gitea.io/gitea/modules/structs"
+)
+
+func listPrimaryLanguageList(ctx *context.APIContext, ownerID int64) {
+	langs, err := repo_model.GetPrimaryRepoLanguageList(ctx, ownerID, ctx.Doer)
+	if err != nil {
+		ctx.InternalServerError(err)
+		return
+	}
+
+	list := []api.LanguageInfo{}
```

Comment:
```suggestion
	list := make([]api.LanguageInfo, len(langs))
```

And use loop index value to set values

Reply:
I would suggest `list := make([]api.LanguageInfo, 0, len(langs))`. We've had problems with initialized arrays used with `append` later which will add elements after the zero-initialized elements.

Reply:
You do not need to use append as there no conditions and it's faster to initialize all elements at once

Reply:
And then you overwrite every initialized element in the loop again. Both versions are nearly identical since some years and I prefer the more readable append style.

Reply:
And I prefer to not review such nit, especially for the non-critical code path.

Pre-allocating or not, it affects nothing, a simple SQL is more than thousands (or maybe millions?) of times slower than the "allocation", so I do not think any one can feel such 0.0001% performance difference.